server/service/individualServices/ReadInventoryData.js

```diff
@@ -1,28 +1,103 @@
-var traceIndicatorIncrementer = 1;
-var mountName = "";
-var ltpStructure = {};
-var requestHeaders = {};
+const IndividualServiceUtility = require('./IndividualServiceUtility');
+const createHttpError = require('http-errors');
 
-exports.readInventoryData = function (_mountName, _ltpStructure, _requestHeaders, _traceIndicatorIncrementer) {
+const FIRMWARE = {
+  MODULE: "firmware-1-0:", COLLECTION: "firmware-collection", CLASS_TYPE: "FIRMWARE_COMPONENT_CLASS_TYPE_PACKAGE",
+  COMPONENT_LIST: "firmware-component-list", PAC: "firmware-component-pac", CAPABILITY: "firmware-component-capability",
+  CLASS: "firmware-component-class", NAME: "firmware-component-name", VERSION: "firmware-component-version", STATUS: "firmware-component-status"
+};
+
+/**
+ * This method performs the set of procedure to gather the inventory data
+ * @param {String} mountName Identifier of the device at the Controller
+ * @param {String} linkId Identifier of the microwave link in the planning
+ * @param {String} uuidUnderTest Identifier of the air-interface under test
+ * @param {Object} ltpStructure ControlConstruct provided from cache
+ * @param {Object} requestHeaders Holds information of the requestHeaders like Xcorrelator , CustomerJourney,User etc.
+ * @param {Integer} traceIndicatorIncrementer traceIndicatorIncrementer to increment the trace indicator
+   @returns {Object} result which contains the inventory data and traceIndicatorIncrementer
+* **/
+exports.readInventoryData = function (mountName, linkId, ltpStructure, uuidUnderTest, requestHeaders, traceIndicatorIncrementer) {
     return new Promise(async function (resolve, reject) {
       try {
-        traceIndicatorIncrementer = _traceIndicatorIncrementer;
-        mountName = _mountName;
-        ltpStructure = _ltpStructure;
-        requestHeaders = _requestHeaders;
-  
-        // code to be implemented
-  
-  
-        let result = {
-          inventory: {},
+      /****************************************************************************************
+      * Declaring required variables
+      ****************************************************************************************/
+        let inventoryResult = {};
+        let inventoryData = {};
+        
+      /****************************************************************************************
+      *  Fetching data for installed-firmware attribute
+      ****************************************************************************************/
+        let installedFirmwareResponse = await RequestForProvidingAcceptanceDataCausesReadingFirmwareList(mountName, requestHeaders, traceIndicatorIncrementer);
+        if(Object.keys(installedFirmwareResponse).length > 0) {
+          if(installedFirmwareResponse.installedFirmware != undefined) {
+            inventoryData.installedFirmware = installedFirmwareResponse.installedFirmware;
```

Comment:
I have used the name "installedFirmware" as key here, as it will be directly put it into response and later in ModifyLowerCamelToKebab function, this shall be directly converted to "installed-firmware" as given in yaml specification. Kindly let me know your views.